vtex-apps/return-app
# feat: add iban frontend validation

#### What problem is this solving?

Add frontend IBAN validation

#### How to test it?

Place a random IBAN it should trigger the error.
[Workspace](https://iban--vtexspain.myvtex.com/account#/my-returns/add)


Before(v2):
![Captura de Pantalla 2022-08-22 a la(s) 11 31 25](https://user-images.githubusercontent.com/96049132/185895871-856329b5-0561-4d54-b98f-2cf6c6e26d0c.png)
![Captura de Pantalla 2022-08-22 a la(s) 11 32 02](https://user-images.githubusercontent.com/96049132/185895876-2c294723-0ee1-4371-ad75-a1342765410f.png)

_Didn't trigger an error because we where only validating if the country code and iban length was correct. But not validating the IBAN structure with mod97 function. If the IBAN is correct this function should returned_ `1`.

Replit:
![Captura de Pantalla 2022-08-22 a la(s) 11 29 29](https://user-images.githubusercontent.com/96049132/185896402-f622a8d9-70c7-4144-9a3b-f3f6160a6ca2.png)

_Here we can see that the behaviour on v2 was wrong. Because this is not a valid number, the mod97 function returned a number different than_ `1`.

After(current implementation):
![Captura de Pantalla 2022-08-22 a la(s) 11 51 57](https://user-images.githubusercontent.com/96049132/185896770-9fd7a640-495b-4901-96dd-efa8c56ea4d1.png)
